usercode/dycore_example/DycoreWrapper.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <iostream>
+#include <map>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <storage/storage-facility.hpp>
+#include "../../wrapper/include/wrappable.h"
+#include "dycore_helper.h"
+
+#include <memory>
+#include "Dycore.h"
+/**
+ * This is just an example to attach to state flags to the storage to allow a consistency check between fortran and cpp
+ */
+template < typename DataStore >
+struct dycore_field {
+    bool cpp_is_uptodate;
+    bool fortran_is_uptodate;
+    DataStore field;
+};
+
+class DycoreWrapper : public wrappable {
+  public:
+    abstract_storage_info get_abstract_storage_info(std::string name, std::vector< int > dims) override {
+        return make_abstract_storage_info(storage_info_t(dims[0], dims[1], dims[2]));
+    }
+
+    void init_optional(std::string name, std::vector< int > dims, bool external_ptr, void *ptr) override {
+        if (m_fields.count(name) == 0) {
+            storage_info_t meta_data(dims[0], dims[1], dims[2]);
+            if (external_ptr) {
+
+                m_fields.emplace(name,
+                    dycore_field< data_store_t >{true,
+                        true,
+                        data_store_t(meta_data, (float_type *)ptr, gridtools::ownership::ExternalCPU, name)});
+                std::cout << "initialized a new gridtools field in pointer sharing mode" << std::endl;
+            } else {
+                m_fields.emplace(name, dycore_field< data_store_t >{true, true, data_store_t(meta_data, name)});
+                std::cout << "initialized a new gridtools field" << std::endl;
+            }
+        }
+    }
+
+    void *get_pointer(std::string name) override {
+        if (m_fields.count(name) > 0) {
+            return (void *)m_fields[name].field.get_storage_ptr()->get_cpu_ptr();
+        } else
+            return nullptr;
+    }
+
+    void notify_push(std::string name) override { m_fields[name].cpp_is_uptodate = true; };
+    void notify_pull(std::string name) override { m_fields[name].fortran_is_uptodate = true; };
+
+    virtual int call(const std::string &action) override {
+        if (action == "init") {
+            std::cout << "Init" << std::endl;
+            // Prepare configuration
+            Configuration config;
+            ConstantFields constantFields;
+            m_dycore = std::make_unique<Dycore>(std::move(config), std::move(constantFields));
```

Comment:
uhmmm, maybe I didnt get this, but why passing an empty config and constant fields? In that case one would just call the default ctr of the m_dycore...

but anyhow, I understand the pattern that you show here, but at least in the shape we see here it has limitations that wont make it work for the cases I think we need. Namely we need to pass arguments to the methods, isnt. The instantiation of the dycore should take some meaningful arguments (other than default constructed config) coming from the caller. I guess the bare minimum is idim, jdim, kdim... How do you pass them form the caller using the call action pattern?